src/Microsoft.Performance.SDK/Options/PluginOption`1.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Performance.SDK.Options;
+
+/// <summary>
+///     Base class for plugin options.
+/// </summary>
+/// <typeparam name="T">
+///     The type of the option value.
+/// </typeparam>
+public abstract class PluginOption<T>
+    : PluginOption
+{
+    private readonly T defaultValue;
+
+    private bool isApplyingDefault;
+
+    private T currentValue;
+
+    /// <summary>
+    ///     Initializes a new instance of the <see cref="PluginOption{T}"/> class.
+    /// </summary>
+    internal PluginOption()
+    {
+        this.IsUsingDefault = true;
+    }
+
+    /// <summary>
+    ///     Gets or initializes the default value of the option. This value is required to be initialized.
+    /// </summary>
+    public required T DefaultValue
+    {
+        get
+        {
+            return this.defaultValue;
+        }
+        init
+        {
+            this.defaultValue = value;
+            this.currentValue = value;
+        }
+    }
+
+    /// <summary>
+    ///     Gets the current value of the option.
```

Comment:
nit: or sets

Reply:
This is a strange case where the getter is public but the setter is internal only. Since plugins can only use the getter, I opted to only document that one.

If you feel strongly about it, I can make an internal method that for setting that is documented accordingly.

Reply:
I'm fine with not documenting.